impl/src/main/java/org/glassfish/json/JsonGeneratorImpl.java

```diff
@@ -487,6 +487,10 @@ protected void writeComma() {
         currentContext.first = false;
     }
 
+    protected boolean inNone() {
+        return currentContext.scope == Scope.IN_NONE;
+    }
```

Comment:
I am not sure that this method is needed. It's simple and there is only one place it called from. Why not moving the logic into if condition in `writeComma()`?

Reply:
The subclass needs to know when we're in the outermost scope so that it knows when it needs to print the newline.

Reply:
Net:  its about the visibility of `currentContext.scope`